### DISCUSSION TITLE
Hooded Sprayer Application

### DISCUSSION THREAD

**Discussion by medlom** [geezacoleman/OpenWeedLocator#40] [General]
Hello Guy
Great work on getting this up and running, it is a great initiative!
What are your thoughts on the possibility of putting an OWL inside a hood? 
I am currently running a CropStalker by Southern Percision Ag and would love to go the next step of only spraying the weeds in the inter-row. 
The hoods are 34cm wide, 40cm high and a length of 60cm. They are  black plastic so artificial light will be required. 
I would like to spray at about 12kph.
Thanks 
Mike

**Comment by geezacoleman** [comment]
Thanks Mike - this was certainly an application we had in mind. It seems quite doable and the artificial lighting means you can tune the detection parameters specifically to the lighting conditions for more reliable performance. With our recent testing 10-12kmh seems very achievable. Having enough space inside the hood is probably the biggest challenge here.

A few things I was thinking of:
* what are the mounting points like inside the shrouds?
* Is it just a single nozzle being activated? If so, you can use a relay control board with only one relay. The code can be simplified too - see some of the discussion in #37.
* where in the enclosure is the nozzle located? Current distance between the camera and nozzle is ~20-25cm though up to 30cm would be better to give the unit more time to spray leading up to the weed.
* 40cm high with the 62 degree field of view from the camera would give you 48cm frame width on the ground, so you would end up with a touch of the sides in the view. Mounting at about 30cm from the ground would give you the ~34cm field of view. Big benefit of this is a much larger on ground resolution, so smaller weeds would be more obvious.

**Reply by medlom** [reply]
There is 10x 50cm area of the stainless steel mount that can be used for mounting inside the shroud. 

It has a single nozzle set back 15cm from front of shroud - not including the nose of the shroud which tapers to a point.
There is another place to mount the nozzle 20cm back from its current position.

The camera might not be able to go much further forward than where the nozzle is currently so without modification I could get 20cm separation but will probably have a chat with manufacturer to see if the nozzle can safely go further back than the back mounting point.

**Reply by geezacoleman** [reply]
Do you have a photo of the inside of one of the shrouds where the unit might mount? It would be possible to make the OWL longer/elongated to fit in the more limited space. How does the inside of the shroud go with mist/dust from the nozzle and crop?

You might be able to get away with 20cm gap with only one nozzle being controlled per unit. There's a bit of code that takes a reasonable amount of time to execute to allocate the detection to a nozzle. But without that gone, the detection -> activation time would be faster.

**Reply by medlom** [reply]
![20220601_145025](https://user-images.githubusercontent.com/83816317/171351252-9206342f-d52f-49db-af5c-de325c7ae238.jpg)
![20220601_145042](https://user-images.githubusercontent.com/83816317/171351264-5e7502a7-0ee0-4dce-8250-5c98100782bb.jpg)

Hopefully these pictures work.
I am not sure about mist and dust inside the hood, the nozzles are coarse to medium spray so hopefully not too misty and seems to be relatively clean inside the hood.

I guess  the solenoid has to be as fast as possible as well?

**Reply by geezacoleman** [reply]
Very helpful thanks! It looks reasonably clean up on the stainless steel plate and closer to the front. 

This could be a quick way to mount it. I've roughly drawn this to scale using that stainless steel plate as 100 x 500 and it seems to fit. The OWL is 125 x 240 so you could most like fit it into the shroud as is. You'd just need to organise a bracket to keep it in there. The power could be routed out through the rear hole. The nozzle would have to be mounted there too.
![image](https://user-images.githubusercontent.com/51358498/171601228-72f8ecae-8c5c-485c-a3e5-12bff7110f0b.png)

For the solenoid - it looks like you're using a Teejet chemsaver nozzle body? These [e-chemsaver solenoids](https://www.ebay.com/itm/384526837273) plug right in where the diaphragm is. So it would be a fairly straightforward switch out. They have fast enough response speeds for spot spraying too. Alternatively, you could use different solenoids/similar to what we used but would need to do some replumbing.

Would have to see how the camera goes with getting dirt on the lens. It looks like the front is cleaner and out the way of mud and spray. If you wanted to chat, flick me an email: guy.coleman[@]sydney.edu.au.
